libs/core/langchain_core/caches.py

```diff
@@ -34,23 +34,23 @@ class BaseCache(ABC):
     """Base interface for cache."""
 
     @abstractmethod
-    def lookup(self, prompt: str, llm_string: str) -> Optional[RETURN_VAL_TYPE]:
+    def lookup(self, prompt: str, llm_string: str, cache_params: Any) -> Optional[RETURN_VAL_TYPE]:
```

Comment:
A few questions:

* What is the type of cache_params? Do you imagine a flat dict would be sufficient?
* What are the semantics for cache_params when doing look up?
* What are the semantics when doing an update?